File: java/src/main/p_1.java
package main;/*
 Find the sum of all the multiples
 of 3 or 5 below 1000
 */

public class p_1 implements util.IProblem {

	public static void main(String[] args) {
		// time function with main.util timing method
		util.time(new p_1());

		// code of main.util timing method but here instead of via main.util class
		long startTime = System.nanoTime();
		new p_1().run();
		long endTime = System.nanoTime();

		long duration = (endTime - startTime);
		
		System.out.println("main.p_1 took " + (float)duration/1000000f + "ms");
	}
	
	public int run() {
		int sum = 0;
		
		for (int i=0; i<1000; i++) {
			if (i % 3 == 0 || i % 5 == 0) {
				sum += i;
			}
		}
		
		// printing the value destroys performance for some reason
		
		System.out.println(sum);
		
		return sum;
	}
}

File: python3/p_1.py
import util


"""
Find the sum of all of the 
multiples of 3 or 5
below 1000
"""


class p_1(util.Problem):
    def __init__(self):
        super().__init__()
        self.problem_id = "1"

    def solve(self):
        return sum(i for i in range(1000) if not i % 3 or not i % 5)


if __name__ == "__main__":
    instance = p_1()
    instance.run()

